src/pip/_internal/resolution/legacy/resolver.py

```diff
@@ -405,6 +404,31 @@ def _find_requirement_link(self, req: InstallRequirement) -> Optional[Link]:
 
         return link
 
+    def _apply_wheel_cache_link(self, req: InstallRequirement) -> None:
+        if not self.wheel_cache:
+            return
+        if not req.link:
+            return
```

Comment:
The function extracts the logic from `_populate_link` and redirects `req.link` to the wheel cache if available. Of something’s not available, no redirection should happen.

Reply:
I'd prefer that we assert this; and fail with an AssertionError instead TBH; but not a blocking concern.